### PR TITLE
feat: add useGTM hook

### PR DESCRIPTION
**Pull request by chambo-e**
A new package with a GTM Provider/hook

This allows us to have the same DX as use-segment and consume gtm in a react way

### CODE REVIEW

File: packages/use-gtm/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "access": "public"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/scaleway/scaleway-lib",
+    "directory": "packages/use-gtm"
+  },
+  "license": "MIT",
+  "peerDependencies": {
+    "react": "17.x"
```
Comment (QuiiBz) [inline packages/use-gtm/package.json:30]: Maybe?
```suggestion
    "react": "17.x || 18.x"
```

**Review by vincentaudebert — APPROVED** [review]
💯

File: packages/use-gtm/package.json (hunk 1 above)
Comment (chambo-e) [reply packages/use-gtm/package.json:30]: I think we'll update all packages once we officially support react 18 :)

Comment (QuiiBz) [reply packages/use-gtm/package.json:30]: Makes sense

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**